presidio-image-anonymizer/tests/test_image_analyzer_engine.py

```diff
@@ -0,0 +1,166 @@
+import pytest
+from presidio_image_anonymizer.image_analyzer_engine import ImageAnalyzerEngine
+from presidio_image_anonymizer.image_recognizer_result import ImageRecognizerResult
+from presidio_analyzer import RecognizerResult
+
+
+@pytest.fixture(scope="function")
+def get_ocr_analyzer_results():
+    ocr_result = {}
+    ocr_result["text"] = [
+        "",
+        "Homey",
+        "Interiors",
+        "was",
+        "created",
+        "by",
+        "Katie",
+        "",
+        "Cromley.",
+    ]
+    ocr_result["left"] = [143, 143, 322, 530, 634, 827, 896, 141, 141]
+    ocr_result["top"] = [64, 67, 67, 76, 64, 64, 64, 134, 134]
+    ocr_result["width"] = [936, 160, 191, 87, 172, 51, 183, 801, 190]
+    ocr_result["height"] = [50, 47, 37, 28, 40, 50, 40, 50, 50]
+    text = " Homey Interiors was created by Katie  Cromley."
+
+    analyzer_result = []
+    analyzer_result.append(RecognizerResult("PERSON", 32, 37, 0.85))
+    analyzer_result.append(RecognizerResult("PERSON", 39, 46, 0.85))
+
+    return ocr_result, text, analyzer_result
+
+
+@pytest.fixture(scope="function")
+def get_image_recognizerresult():
+    results = []
+    results.append(ImageRecognizerResult("PERSON", 32, 37, 0.85, 896, 64, 183, 40))
+    results.append(ImageRecognizerResult("PERSON", 39, 46, 0.85, 141, 134, 190, 50))
+    return results
+
+
+def test_given_empty_dict_then_get_text_from_ocr_dict_returns_empty_str():
+    ocr_result = {}
+    expected_text = ""
+    text = ImageAnalyzerEngine.get_text_from_ocr_dict(ocr_result)
+
+    assert expected_text == text
+
+
+@pytest.mark.parametrize(
+    "sep, expected_text",
+    [
+        (" ", " Homey Interiors was created by Katie  Cromley."),
+        ("+", "+Homey+Interiors+was+created+by+Katie++Cromley."),
+    ],
+)
+def test_given_valid_dict_then_get_text_from_ocr_dict_returns_correct_str(
+    get_ocr_analyzer_results, sep, expected_text
+):
+    ocr_result, t, a = get_ocr_analyzer_results
+    text = ImageAnalyzerEngine.get_text_from_ocr_dict(ocr_result, sep)
+    assert expected_text == text
+
+
+def test_given_wrong_keys_in_dict_then_get_text_from_ocr_dict_returns_exception():
+    ocr_result = {"words": ["John"], "level": [0]}
+    expected_exception_message = "Key 'text' not found in dictionary"
+    with pytest.raises(KeyError) as e:

```

Comment:
with pytest.raises(KeyError, match="Key 'text' not found in dictionary"):

without the assertion

Reply:
changed as per suggestion